### ISSUE TITLE
Should header definitions be allowed to be overridden at the resource level?

### ISSUE THREAD

**Issue by infomofo** [gilt/apidoc#287]
For example, if I have a token that is required for PUT operations but not GET operations

**Comment by mbryzek** [comment]
Gut feeling is this seems weird. Headers feel like something I'd want to setup up front for how to interact with a service. Everything else for a particular resource would then be part of the request.

What's your use case?

**Comment by infomofo** [comment]
It goes back to something paul had suggested on hipchat- using a custom header to supplement our delete operation.  We actually decided not to go this route, but I was just curious how it would be implemented, as it is a header we'd only want for this particular operations and we wouldn't need it for any of the other resources on the client.

The swagger spec does allow specification of headers as parameters for each resource, although I'm not sure how this is implemented.

**Comment by mbryzek** [comment]
yeah - i think what we would have to do is to expose a "headers" parameter on each method call. That parameter might conflict with user defined parameters - so it might need to be called "_headers" or some other weird variation (or we use headers unless the name is already taken).

That might actually be a nice change to the clients - I can see us standardizing these 'request' parameters then accepting them when you create the client and also on all of the individual methods.

**Comment by infomofo** [comment]
Even in the headers current form, I can't figure out how the client specifies values for them using the generated play 2.3 client.  Here's my apidoc: http://www.apidoc.me/codalytic/codalytic/latest
